crates/sui-core/src/signature_verifier.rs

```diff
@@ -320,17 +325,27 @@ impl SignatureVerifier {
     }
 
     pub fn verify_tx(&self, signed_tx: &SenderSignedData) -> SuiResult {
-        self.signed_data_cache
-            .is_verified(signed_tx.full_message_digest(), || {
+        self.signed_data_cache.is_verified(
+            signed_tx.full_message_digest(),
+            || {
                 signed_tx.verify_epoch(self.committee.epoch())?;
                 let jwks = self.jwks.read().clone();
-                let aux_data = VerifyParams::new(
+                let verify_params = VerifyParams::new(
                     jwks,
                     self.zk_login_params.supported_providers.clone(),
                     self.zk_login_params.env.clone(),
                 );
-                signed_tx.verify_message_signature(&aux_data)
-            })
+                if signed_tx.has_zklogin_sig() {
+                    signed_tx.verify_message_signature_with_cache(
```

Comment:
1. IIUC, in the case of multisig, we only cache today the full sig, and not also the internal individual signatures. I think in this PR we can continue with that approach, and deal with multisig later (as it's less common).
2. What about having a function like `verify_reusable_signature()` or `verify_aux_data` in signed_tx, which will be empty for all authenticators but zk, and in the latter it will verify the zk proof. Also, a function like `verify_message_signature_withoutaud_data()`, and have `verify_message_signature` call both of those. then we could leave the cache here and just call the above functions.

Reply:
this seems like the right approach to me. Basically, break it into two pieces:

- check that the proof is valid, either via the cache or via proof verification
- then check that the ephemeral signature is valid

Reply:
that still requires adding the verify_message_signature to include cache right? 

```
    fn verify_message_signature(&self, verify_params: &VerifyParams) -> SuiResult;
```
to 

```
    fn verify_message_signature(&self, verify_params: &VerifyParams, cache: &VerifiedDigestCache<ZKLoginInputsDigest>,) -> SuiResult;
```